node/services/service.go

```diff
@@ -161,13 +161,44 @@ func DASer(
 	return das
 }
 
-// FraudService constructs fraud proof service.
+// FraudService constructs fraud proof service with disabled syncer.
 func FraudService(
+	lc fx.Lifecycle,
 	sub *pubsub.PubSub,
+	host host.Host,
 	hstore header.Store,
 	ds datastore.Batching,
-) fraud.Service {
-	return fraud.NewService(sub, hstore.GetByHeight, ds)
+) (fraud.Service, error) {
+	return newFraudService(lc, sub, host, hstore, ds, false)
+}
+
+// FraudServiceWithSyncer constructs fraud proof service with enabled syncer.
+func FraudServiceWithSyncer(
+	lc fx.Lifecycle,
+	sub *pubsub.PubSub,
+	host host.Host,
+	hstore header.Store,
+	ds datastore.Batching,
+) (fraud.Service, error) {
+	return newFraudService(lc, sub, host, hstore, ds, true)
+}
+
+func newFraudService(
+	lc fx.Lifecycle,
+	sub *pubsub.PubSub,
+	host host.Host,
+	hstore header.Store,
+	ds datastore.Batching,
+	isEnabled bool) (fraud.Service, error) {
+	pservice := fraud.NewProofService(sub, host, hstore.GetByHeight, ds, isEnabled)
```

Comment:
either `service` or `proofService` is better than `pservice`. I would try to not abbreviate local variable names very often unless they are very long or redundant

Reply:
It's a common practice in Go and Go std libs, though, especially when the constructor already tells you the full name of the struct we work with.